### ISSUE TITLE
Extension list in argument parser gets really long with compressions

### ISSUE THREAD

**Issue by h-2** [seqan/seqan#1830]
e.g:
```
    -q, --query IN
          Query sequences. Valid filetypes are: .fq, .fq.bgzf, .fq.gz, .fq.bz2, .fastq, .fastq.bgzf, .fastq.gz,
          .fastq.bz2, .fa, .fa.bgzf, .fa.gz, .fa.bz2, .fasta, .fasta.bgzf, .fasta.gz, .fasta.bz2, .faa, .faa.bgzf,
          .faa.gz, .faa.bz2, .ffn, .ffn.bgzf, .ffn.gz, .ffn.bz2, .fna, .fna.bgzf, .fna.gz, .fna.bz2, .frn, .frn.bgzf,
          .frn.gz, .frn.bz2, .embl, .embl.bgzf, .embl.gz, .embl.bz2, .gbk, .gbk.bgzf, .gbk.gz, .gbk.bz2, .raw,
          .raw.bgzf, .raw.gz, and .raw.bz2.
```
Could we instead display something like:
```
    -q, --query IN
          Query sequences. Valid filetypes are: .fq[.bgzf|.gz|.bz2], .fastq[.bgzf|.gz|.bz2], .fa[.bgzf|.gz|.bz2], 
          .fasta[.bgzf|.gz|.bz2], .faa[.bgzf|.gz|.bz2], .ffn[.bgzf|.gz|.bz2], .fna[.bgzf|.gz|.bz2], .frn[.bgzf|.gz|.bz2],
          .embl[.bgzf|.gz|.bz2], .gbk[.bgzf|.gz|.bz2], .raw[.bgzf|.gz|.bz2]
```

Or:
```
    -q, --query IN
          Query sequences. Valid filetypes are: .fq, .fastq, .fa, .fasta, .faa, .ffn, .fna, .frn, .embl, .gbk, .raw
          Additionally any of the listed extensions may be extended by .bgzf, .gz or .bz2 for transparent 
          (de)compression.
```

The last approach looks nicer, but it does have the problem that sometimes not all extensions have all compression formats, e.g. `.bam` cannot by extended by anything, because it already is compressed.

@rrahn @smehringer what do you think?

**Comment by smehringer** [comment]
Do we always allow compressed files and want the compressed file extensions always to be shown or does the user needs to specify compression formats (/for each file format)?

**Comment by rrahn** [comment]
Well, the compression formats will be automatically depicted, if the corresponding compression library is activated in seqan, e.g., SEQAN_HAS_ZLIB. I think the compression should be applied whenever possible.

@h-2 would it not be possible to let the bgzf_buffer read from a gzipped stream? I think technically, this would be possible, though I am not sure.
Anyway, I suggest a compromise between solution **b** and **c**.
Something like 
```
-q --query IN
    Query sequences. Valid filetypes are: .sam[.x], .bam,
    where x is any of the following extensions  .bgzf, .gz or .bz2 for transparent 
    (de)compression.
```

**Comment by smehringer** [comment]
@h-2 your example now looks like this:

```
    -i, --input IN
          input file. Valid filetypes are: .raw[.x], .gbk[.x], .frn[.x], .fq[.x], .fna[.x], .ffn[.x], .fastq[.x],
          .fasta[.x], .faa[.x], .fa[.x], and .embl[.x], where x is any of the following extensions: gz, bz, and bgzf
          for transparent (de)compression.

```

**Comment by h-2** [comment]
fixed by #1843
